hs_core/tasks.py

```diff
@@ -344,34 +345,40 @@ def create_bag_by_irods(resource_id):
 
     bag_path = res.bag_path
 
-    metadata_dirty = istorage.getAVU(res.root_path, 'metadata_dirty')
+    metadata_dirty = res.getAVU('metadata_dirty')
+    metadata_dirty = metadata_dirty is None or metadata_dirty
     # if metadata has been changed, then regenerate metadata xml files
-    if metadata_dirty is None or metadata_dirty.lower() == "true":
+    if metadata_dirty:
         create_bag_metadata_files(res)
 
-    irods_bagit_input_path = res.get_irods_path(resource_id, prepend_short_id=False)
-
-    # only proceed when the resource is not deleted potentially by another request
-    # when being downloaded
-    is_exist = istorage.exists(irods_bagit_input_path)
-    if is_exist:
+    bag_modified = res.getAVU("bag_modified")
+    bag_modified = bag_modified is None or bag_modified
+    if metadata_dirty or bag_modified:
         create_bagit_files_by_irods(res, istorage)
-        try:
-            # call iRODS run and ibun command to zip the bag, ignore SessionException
-            # for now as a workaround which could be raised from potential race conditions when
-            # multiple ibun commands try to create the same zip file or the very same resource
-            # gets deleted by another request when being downloaded
-            istorage.zipup(irods_bagit_input_path, bag_path)
-            if res.raccess.published:
-                # compute checksum to meet DataONE distribution requirement
-                chksum = istorage.checksum(bag_path)
-                res.bag_checksum = chksum
-            istorage.setAVU(irods_bagit_input_path, 'bag_modified', "false")
-            return res.bag_url
-        except SessionException as ex:
-            raise SessionException(-1, '', ex.stderr)
-    else:
-        raise ObjectDoesNotExist('Resource {} does not exist.'.format(resource_id))
+        res.setAVU("bag_modified", False)
```

Comment:
I am wondering whether metadata_dirty flag only needs to be set to True here since create_bagit_files_by_irods() also updates metadata file?

Reply:
create_bag_metadata_files creates the metadata files above and sets the flag when it finishes.  https://github.com/hydroshare/hydroshare/pull/4114/files/fdc0a8656e9b57d7b03f4a5e1a1c12580889683b#diff-182216ce9ee2cc3e1f410a911873b12dc126f5e025f7b81f8d5d1ecdfe234b0bR352